cookbooks/aws-parallelcluster-environment/resources/nfs/nfs_ubuntu22+.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+provides :nfs, platform: 'ubuntu' do |node|
+  node['platform_version'].to_i >= 22
+end
+
+unified_mode true
+
+use 'partial/_install_nfs_debian'
+use 'partial/_install_nfs4_and_disable'
+use 'partial/_configure'
+
+default_action :setup
+
+action :setup do
+  action_install_nfs
+  action_install_nfs4
+  action_disable_start_at_boot
+  node.default['nfs']['config']['server_template'] = '/etc/nfs.conf.d/nfs.conf'
```

Comment:
You should not do this, you should instead upgrade NFS third party cookbook: https://supermarket.chef.io/cookbooks/nfs#changelog

Support for ubuntu has been added on 5.0.1 and we have 5.0.0 

Maybe with this change we don't need to do any customization at all.

Reply:
5.0.1 fails due to this known issue and our workaround isn't helping in Ubuntu 22 for some reason https://github.com/sous-chefs/nfs/issues/93